### PR TITLE
Submit batch to contract

### LINKED ISSUE #6: T3.2 Replace call to IntrinsicGas
Parent task #4

### LINKED ISSUE #7: T3.3 Update BatcherHash
Parent task #4

### PR DESCRIPTION
**Pull request by nyospe**
Addresses https://github.com/EspressoSystems/op-leader-election/issues/6
Addresses https://github.com/EspressoSystems/op-leader-election/issues/7

### CODE REVIEW

File: op-node/chaincfg/chains.go [hunk 1]
```diff
@@ -97,6 +99,7 @@ var Sepolia = rollup.Config{
 	BatchInboxAddress:      common.HexToAddress("0xff00000000000000000000000000000011155420"),
 	DepositContractAddress: common.HexToAddress("0x16fc5058f25648194471939df75cf27a2fdc48bc"),
 	L1SystemConfigAddress:  common.HexToAddress("0x034edd2a225f7f429a63e0f1d2084b9e0a93b538"),
+	BatchInboxContractAddr: common.HexToAddress("0xff00000000000000000000000000000011155420"),
```
Comment (philippecamacho) [inline op-node/chaincfg/chains.go]: Why are the addresses of the addresses of the Batch Inbox and Batch Inbox Contract the same?

File: packages/contracts-bedrock/deploy-config/hardhat.json [hunk 2]
```diff
@@ -38,6 +39,11 @@
   "numDeployConfirmations": 1,
   "eip1559Denominator": 50,
   "eip1559Elasticity": 10,
+<<<<<<< HEAD
```
Comment (philippecamacho) [inline packages/contracts-bedrock/deploy-config/hardhat.json]: Looks like some merge conflict is pending here.

File: packages/contracts-bedrock/deploy-config/internal-devnet.json [hunk 3]
```diff
@@ -2,51 +2,48 @@
...
   "governanceTokenOwner": "0x858F0751ef8B4067f0d2668C076BDB50a8549fbF",
-
   "l2GenesisBlockGasLimit": "0x17D7840",
   "l2GenesisBlockBaseFeePerGas": "0x3b9aca00",
-
-
   "gasPriceOracleOverhead": 2100,
   "gasPriceOracleScalar": 1000000,
   "eip1559Denominator": 50,
+<<<<<<< HEAD
```
Comment (philippecamacho) [inline packages/contracts-bedrock/deploy-config/internal-devnet.json]: Merge conflict to be resolved.

File: packages/contracts-bedrock/deploy-config/sepolia.json [hunk 4]
```diff
@@ -38,6 +39,11 @@
   "l2GenesisBlockBaseFeePerGas": "0x3b9aca00",
   "eip1559Denominator": 50,
   "eip1559Elasticity": 6,
+<<<<<<< HEAD
```
Comment (philippecamacho) [inline packages/contracts-bedrock/deploy-config/sepolia.json]: Merge conflict to be resolved.

File: op-node/chaincfg/chains.go (hunk 1 above)
Comment (nyospe) [reply op-node/chaincfg/chains.go]: Placeholder. We need to publish your contract to different chains before this goes live on each.

File: packages/contracts-bedrock/deploy-config/hardhat.json (hunk 2 above)
Comment (nyospe) [reply packages/contracts-bedrock/deploy-config/hardhat.json]: It's very odd that I haven't been able to find these when searching in VSCode... I ended up having to use ripgrep to find them.

File: packages/contracts-bedrock/deploy-config/internal-devnet.json [hunk 5]
```diff
@@ -2,51 +2,42 @@
...
-
-
   "gasPriceOracleOverhead": 2100,
   "gasPriceOracleScalar": 1000000,
   "eip1559Denominator": 50,
   "eip1559Elasticity": 10,
-
   "leaderElectionMaxParticipants": 5
-}
+}
```
Comment (sveitser) [inline packages/contracts-bedrock/deploy-config/internal-devnet.json]: Is there a particular reason why the newlines are removed at the end of all the JSON files with changes?

File: op-batcher/batcher/driver.go [hunk 6]
```diff
@@ -115,6 +122,13 @@ func NewBatchSubmitter(ctx context.Context, cfg Config, l log.Logger, m metrics.
 
 	cfg.metr = m
 
+	if cfg.BatchInboxAbi == nil {
+		biAbi, err := bindings.LeaderElectionBatchInboxMetaData.GetAbi()
+		if err == nil && biAbi != nil {
+			cfg.BatchInboxAbi = biAbi
+		}
```
Comment (sveitser) [inline op-batcher/batcher/driver.go:136]: Should there be a `TODO` comment here or is it okay to continue with `BatchInboxAbi` unset?

File: packages/contracts-bedrock/deploy-config/internal-devnet.json (hunk 5 above)
Comment (nyospe) [reply packages/contracts-bedrock/deploy-config/internal-devnet.json]: Apparently I have a visual studio plug-in badly formatting my json, and I can't figure out which one is doing it. I'll try to fix the files in vim...

File: op-batcher/batcher/driver.go (hunk 6 above)
Comment (nyospe) [reply op-batcher/batcher/driver.go:136]: I'm not actually sure what should happen here. I need this to get the function selector for the contract, and it should always exist, but...

Comment (sveitser) [reply op-batcher/batcher/driver.go:136]: If we crash if it's not set does that break anything?

**Review by sveitser — APPROVED** [review]
LGTM

**Pull request merged**